internal/provider/customer_resource.go

```diff
@@ -374,3 +345,34 @@ func (r *CustomerResource) Delete(ctx context.Context, req resource.DeleteReques
 func (r *CustomerResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
+
+func getCustomerResourceModelFromCustomer(appID string, customer *rtypes.Customer) CustomerResourceModel {
+	entitlements := make(map[string]string)
+
+	for _, entitlement := range customer.Entitlements {
+		entitlements[entitlement.Name] = entitlement.Value
+	}
+
+	entitlementValues, _ := types.MapValueFrom(context.Background(), types.StringType, entitlements)
+
+	return CustomerResourceModel{
+		Id:                               types.StringValue(fmt.Sprintf("app/%s/customer/%s", appID, customer.ID)),
+		AppId:                            types.StringValue(appID),
+		ChannelId:                        types.StringValue(customer.Channels[0].ID),
+		Email:                            types.StringValue(customer.Email),
+		EntitlementValues:                entitlementValues,
+		ExpiresAt:                        types.StringValue(customer.Expires.Format("2006-01-02T15:04:05Z")),
```

Comment:
Will this work if the customer doesn't have an expiration date?

Reply:
Thanks for catching that. I've updated this and added a test for it.